CI: "rebase atop of the latest target branch" failing

On CI job [kata-containers-ci-on-push / run-kata-coco-tests / run-k8s-tests-on-tdx (qemu-tdx)](https://github.com/kata-containers/kata-containers/actions/runs/6641634670/job/18052687864?pr=7231#logs) I'm getting the following error when it tried to rebase my last changes:

```
Adding user name and email to the local git repo
Rebasing atop of the latest main
fatal: It seems that there is already a rebase-merge directory, and
I wonder if you are in the middle of another rebase.  If that is the
case, please try
	git rebase (--continue | --abort | --skip)
If that is not the case, please
	rm -fr ".git/rebase-merge"
and run me again.  I am stopping in case you still have something
valuable there.

Error: Process completed with exit code 128.
```

The TDX job run on bare-metal machine so likely there is left rebase on the system.

I don't know why it end up on that situation.

--- a/tests/git-helper.sh
+++ b/tests/git-helper.sh
@@ -19,6 +19,8 @@ function add_kata_bot_info() {
 function rebase_atop_of_the_latest_target_branch() {
 	if [ -n "${TARGET_BRANCH}" ]; then
 		echo "Rebasing atop of the latest ${TARGET_BRANCH}"
+		# Recover from any previous rebase left halfway
+		git rebase --abort 2> /dev/null || true
 		git rebase origin/${TARGET_BRANCH}
 	fi
 }

